UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/WebhookController.cs

```diff
@@ -30,12 +43,49 @@ public IActionResult NewFilesPublishedOptions()
         [Route("/webhook/newfilespublished")]
         public virtual async Task<IActionResult> NewFilesPublished()
         {
-            ////Adding payload to log temporarily so that in can be used for unit testing etc. 
-            using var reader = new StreamReader(Request.Body, Encoding.UTF8);
+            using StreamReader reader = new(Request.Body, Encoding.UTF8);
             string payload = await reader.ReadToEndAsync();
-            Logger.LogInformation(EventIds.ClearFSSSearchCacheEventStart.ToEventId(), "Clear FSS Search Cache Event started for _X-Correlation-ID:{correlationId} Payload: {payload}", GetCurrentCorrelationId(), payload);
 
-            return GetCacheResponse(); 
+            _logger.LogInformation(EventIds.ClearFSSSearchCacheEventStarted.ToEventId(), "Clear FSS search cache event started for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+            if (string.IsNullOrEmpty(payload))
+            {
+                _logger.LogInformation(EventIds.ClearFSSSearchCacheValidationEvent.ToEventId(), "Payload is null or empty for Enterprise event _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                return GetCacheResponse();
+            }
+            EventGridEvent eventGridEvent = new();
+
+            JsonConvert.PopulateObject(payload, eventGridEvent);
+            if (eventGridEvent.Data == null || eventGridEvent.Data.ToString() == "{}")
+            {
+                _logger.LogInformation(EventIds.ClearFSSSearchCacheValidationEvent.ToEventId(), "Payload data is null or empty for Enterprise event _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
+                return GetCacheResponse();
+            }
+            FSSNewFilesPublishedEventData data = (eventGridEvent.Data as JObject).ToObject<FSSNewFilesPublishedEventData>();
+
+            
+
+            _logger.LogInformation(EventIds.ClearFSSSearchCacheEventStarted.ToEventId(), "Enterprise event data deserialized. Data:{data} and _X-Correlation-ID:{correlationId}", JsonConvert.SerializeObject(data), GetCurrentCorrelationId());
+
+            FluentValidation.Results.ValidationResult validationResult = await _webhookService.ValidateNewFilesPublishedEventData(data);
+
+            string productType = data.Attributes.Where(a => a.Key == "Product Type").Select(a => a.Value).FirstOrDefault();
+
+            if (!validationResult.IsValid)
+            {
+                _logger.LogInformation(EventIds.ClearFSSSearchCacheValidationEvent.ToEventId(), "Required attributes missing in event data from Enterprise event for clear FSS search cache from Azure table for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
```

Comment:
I'm wondering is this should be a `LogError` as it's something that should not happen.

Reply:
Corrected